11001.md

```diff
@@ -0,0 +1,90 @@
+NIP-XXXX
+======
+
+messages specification for peer 2 peer
+--------------------------------------
+
+`draft` `optional` `author:XXXX`
+
+Peer-to-peer is one of the most important applications needed in the Bitcoin space.
+
+Nostr can enhance liquidity by enabling existing profiles to use their reputation to trade with other peers. Reputation is a key element that makes large P2P platforms like Paxful or Binance P2P easy to use and scale, because it's easy to find someone who is likely reliable. The drawback is that identity in those platforms is locked in, and reputation built on one of those platforms can't be reused on another. Using Nostr would solve that reputation problem.
+
+Users could also choose to create a one-time identity and participate on the same based platform by simply creating a new key-pair for each trade. It would make it possible to merge currently separate use cases where platforms today are either relying on bonds (like RoboSats) or reputation (like Paxful, Binance P2P, HodlHodl).
+
+An ecosystem where different clients use the same specification on how to create and settle trades would create a thriving environment where clients compete in their UX and product offerings, to the benefit of the users and the global liquidity of the network.
+
+While it would be likely that clients are initially dedicated to this application, an ideal scenario would be that, over time, generic clients also integrate this NIP. This would incentivize a large group of users who would own Bitcoin, from zaps for instance, to trade with each other for local fiat currencies, without having the current barrier to entry of having a dedicated app.
+
+## Data structure
+
+an event looks like this:
+
+```
+let event = {
+  kind: 11001,
+  created_at: Math.floor(Date.now() / 1000),
+  tags: [
+    ["D", "BUY"],
+    ["C", "BTC/NGN"],
+    ["P", "10000000.00000000"], // price, 1 BTC = 10000000 NGN
```

Comment:
I think this may be unpractical. It is better to have the price as a percentage difference from the price given by some exchange API, as Bisq does, for example.

Reply:
Why not support both. P could have a type flag that's either "static" or "dynamic". If it's dynamic then an endpoint is expected

Reply:
> Why not support both.

I guess the main reason to converge to a single solution is that the more complex the spec is, the more cumbersome it will be for clients to build on top of it.

> I think this may be unpractical. It is better to have the price as a percentage difference from the price given by some exchange API, as Bisq does, for example.

I think the solution @fiatjaf suggest would put more burden on the order `taker`, as opposed to the `maker` (which can typically be a market market, that could automate the price update using a replace `e` tag). 

Also it would add complexity to the client because now client have to fetch price on the exchange (and manage all the use case like if the exchange doesn't respond) in addition to the nostr websocket connection. 

My current thinking is that it should be optional for the client to compare the actual price than what is present on some  exchange, but this comparison should be optional and serve as a reference. I think Paxful does that (show the premium/discount) but I need to double check.